python/ray/_raylet.pyx

```diff
@@ -338,6 +335,11 @@ class StreamingObjectRefGenerator:
         ready, unready = await asyncio.wait([ref], timeout=timeout_s)
         if len(unready) > 0:
             return ObjectRef.nil()
+        # NOTE(swang): Hack to avoid asyncio warnings about not retrieving the
+        # exception. The exception will get returned to the user through the
+        # below code.
+        for result in ready:
+            exc = result.exception()
```

Comment:
This will suppress any form of unexpected exception. I would suggest to explicitly wrap the ref in a task instead and ignore the specific exception type:
```python
async def peek_stream_wrapper(ref: "ImNotSureWhatTypeThisIs"):
    try:
        await ref
    except ObjectRefStreamEndOfStreamError:
        pass # Expected.

peek_stream_task = asyncio.create_task(
    peek_stream_wrapper(core_worker.peek_object_ref_stream(self._generator_ref))
)
ready, unready = await asyncio.wait([peek_stream_task], timeout=timeout_s)
```

`asyncio.wait` does this under the hood anyways so there shouldn't be additional overhead

Reply:
Hmm actually I don't think we should do this. This part of the code is only meant to wait for the ref to become ready; we just don't have a way to do this for asyncio right now without also fetching the data.

The expected use case is that the user will then do something with the returned ref once it's ready. I believe Ray also warns if the ref has an exception and is never read, so we'll be warning twice if we don't suppress the asyncio error.

Reply:
Ah I see so this could raise other exceptions aside from `ObjectRefStreamEndOfStreamError`?

Ideally we would "whitelist" the expected exceptions here -- maybe a conservative way to do it would be `except RayError`? I'm just a little uneasy about suppressing all exceptions, there may be some corner case that raises an unexpected exception and identifying/debugging it will be much easier if the exception is logged.

Ultimately up to you though, if you don't think this is an issue then suppressing them all is fine. Though it's probably still slightly less hacky to wrap it in a task and catch bare `Exception`.

Reply:
Yeah, to clarify, the reasoning for suppressing all errors is that at this point, the user hasn't actually tried to receive the error yet. This function is supposed to return the `ObjectRef` that the user can await, and then at point, asyncio will warn if the user doesn't retrieve the exception.

So either we don't suppress the error here, and the user will always see an extra warning, or we do suppress the error here, and then Ray will warn if the user doesn't do anything with the ObjectRef.

Reply:
Got it. Thanks for clarifying.